version: "3.8"
services:
  db:
    image: mysql:8.0
    restart: always
    env_file:
      - .env
    environment:
      TZ: ${TZ}
    ports:
      - "3308:3306"
    volumes:
      - pgdata:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10
    networks:
      - ices-network

  web:
    build: .
    command:
      [
        "/bin/sh",
        "-c",
        "sleep 5 && alembic upgrade head && python -m app.core.create_admin && uvicorn app.main:app --host 0.0.0.0 --port 5000",
      ]
    env_file:
      - .env
    environment:
      DATABASE_URL: mysql+pymysql://root:root@db:3306/ices_db
      TZ: ${TZ}
      SECRET_KEY: ${SECRET_KEY}
    depends_on:
      db:
        condition: service_healthy
    ports:
      - "5000:5000"
    volumes:
      - .:/code
    networks:
      - ices-network

volumes:
  pgdata:

networks:
  ices-network:
    driver: bridge
